gnuradio-runtime/lib/math/random.cc

```diff
@@ -71,6 +76,28 @@ namespace gr {
     if(seed==0) d_seed = static_cast<unsigned int>(std::time(0));
     else d_seed = seed;
     d_rng->seed(d_seed);
+    // reinstantiate generators. Otherwise reseed doesn't take effect.
+    delete d_generator;
```

Comment:
Both d_generator and d_integer_generator are set to NULL in the constructor, then this method (reseed) is called.  Here then you call delete on NULL pointers.